[DOCS-1700] Update docs for automation history (#1803)
wandb/docs: models/automations/automation-events.mdx

<EnterpriseCloudOnly/>
</Info>

> An automation can start when a specific event occurs within a project or registry. Learn more about automations in the [Automations overview](/models/automations/) or [Create an automation](/models/automations/create-automations/).

Author: ```suggestion
An automation starts when a specific event occurs within a project, registry, or collection. For example, when an artifact version is created, aliased, tagged, or linked, when a run's status changes, or when a run metric meets or changes by a given threshold. Learn more about automations in the [Automations overview](/models/automations/) or [Create an automation](/models/automations/create-automations/).
```

| Action                                           | Registry scope | Project scope |
|--------------------------------------------------|----------------|---------------|
| [Artifact version](#artifact-events) linked      | ✓              |               |
| [Artifact version](#artifact-events) aliased     | ✓              | ✓             |
| [Artifact version](#artifact-events) created     |                | ✓             |

> | [Artifact version](#artifact-events) tagged      |                | ✓             |

Reviewer: **suggestion:** "Artifact version created" is only supported in a collection scope (not project), so you may need to add a `Collection scope` column if you're keeping this change.

FWIW, you may also want to consider splitting the changes into at least 2 separate PRs, to unblock all the changes more easily:
- one for automations history edits
- one or more for other automation content edits

It's also worth noting these boundaries are more strictly curated in the UI.  Users who create/edit automations through the API will find that the table looks more like this:

```markdown
| Action                                           | Registry | Project | Collection 
|--------------------------------------------------|----------|---------|------------
| [Artifact version](#artifact-events) linked      | ✓        | ✓       | ✓
| [Artifact version](#artifact-events) aliased     | ✓        | ✓       | ✓
| [Artifact version](#artifact-events) created     |          |         | ✓
| [Artifact version](#artifact-events) tagged      | ✓        | ✓       | ✓
```

Author: This table was in response to feedback that the previous content was misleading because it was too bifurcated (maybe from the previous PR). I've adjusted these changes further, and split out to a separate PR.

| [Run status](#run-status-change) changed         |                | ✓             |

> | [Run metric](#run-metrics-change) threshold met  |                | ✓             |

Author: ```suggestion
```

## Artifact events
In a particular registry, an automation can start when an artifact version is linked or aliased. Automations in a given registry apply to all of the registry's collections.

In a project, an automation can start when an artifact version is created, tagged, or aliased. Project automations are scoped to the team and apply to all of the project's collections.

Regular expressions are supported for aliases and tags.

## Run status change
<Note>
Currently available only in [W&B Multi-tenant Cloud](/platform/hosting/#wb-multi-tenant-cloud).
</Note>

> In a project, an automation can start when a run changes its [status](/models/runs/#run-states) to **Running**, **Finished**, or **Failed**. An automation cannot start when a run's status is **Killed**. This status indicates that the run was stopped forcibly by an admin user.

Reviewer: **question:** Have we confirmed the detail about `killed` status?  I'm actually not sure if `killed` is handled similarly to `crashed` by the backend (`run crashed` will trigger for any `run failed` event, but haven't been able to test the behavior for `run killed` yet)

Author: That fact is not new in this PR, it came with run status notifications docs, and was a correction from Stephen. https://github.com/wandb/docs/commit/0258b4373345b701ccf2ccae1c4dba296af30a73 It's looking like it's new because it is moved from elsewhere in the content. I suppose it could have changed in the meantime, but nobody has told me.

| Action                                           | Registry scope | Project scope |
|--------------------------------------------------|----------------|---------------|
| [Artifact version](#artifact-events) linked      | ✓              |               |
| [Artifact version](#artifact-events) aliased     | ✓              | ✓             |
| [Artifact version](#artifact-events) created     |                | ✓             |
| [Artifact version](#artifact-events) tagged      |                | ✓             |
| [Run status](#run-status-change) changed         |                | ✓             |
| [Run metric](#run-metrics-change) threshold met  |                | ✓             |

## Artifact events
In a particular registry, an automation can start when an artifact version is linked or aliased. Automations in a given registry apply to all of the registry's collections.

In a project, an automation can start when an artifact version is created, tagged, or aliased. Project automations are scoped to the team and apply to all of the project's collections.

Regular expressions are supported for aliases and tags.

## Run status change
<Note>

> - Currently available only in [W&B Multi-tenant Cloud](/platform/hosting/#wb-multi-tenant-cloud).
> - A run with **Killed** status cannot trigger an automation. This status indicates that the run was stopped forcibly by an admin user.

Author: ```suggestion
Currently available only in [W&B Multi-tenant Cloud](/platform/hosting/#wb-multi-tenant-cloud).
```

</Note>

> In a project, an automation can start when a run changes its [status](/models/runs/#run-states) to **Running**, **Finished**, or **Failed**. An automation cannot start when a run's status is **Killed**.

Author: ```suggestion
In a project, an automation can start when a run changes its [status](/models/runs/#run-states) to **Running**, **Finished**, or **Failed**. An automation cannot start when a run's status is **Killed**. This status indicates that the run was stopped forcibly by an admin user.
```
